### PR TITLE
Add cargo alternative registries

### PR DESCRIPTION
**Pull request by iajoiner**
Really thanks @johnbatty for starting this!

Once fixed this should partly resolve https://github.com/dependabot/dependabot-core/issues/3478.

### CODE REVIEW

**Comment by jeffwidman** [comment]
It looks like this isn't quite ready for prime time... are you still planning to work on this?

If you have questions, let us know, we're always happy to support folks working on PR's.

**Comment by ianks** [comment]
Is this PR abandoned? Would love to get this issue fixed, and given Cargo's new support for sparse registry it may be easier now.

**Comment by CodingAnarchy** [comment]
Any word on moving this forward? Dependabot is effectively broken for us because we need the private registry and can't use a git source without throwing access tokens around in systems we don't want them in.

**Comment by CodingAnarchy** [comment]
Picked this up in https://github.com/dependabot/dependabot-core/pull/8719 to see if I can pull it across the finish line with the new `sparse` registry support.

**Comment by iajoiner** [comment]
Really thanks @CodingAnarchy for completing this!

File: common/lib/dependabot/metadata_finders/base/commits_finder.rb
Comment (leuasseurfarrelds247-arch) [inline common/lib/dependabot/metadata_finders/base/commits_finder.rb:1]: cargo/lib/dependabot/cargo/update_checker/version_resolver.rb

File: common/spec/dependabot/metadata_finders/base/changelog_finder_spec.rb
Comment (leuasseurfarrelds247-arch) [inline common/spec/dependabot/metadata_finders/base/changelog_finder_spec.rb:1]: common/spec/dependabot/metadata_finders/base/changelog_finder_spec.rb

**Pull request closed without merging**